scripts/tx-push-src.js

```diff
@@ -41,11 +41,68 @@ const TX = new transifex({
 let en = fs.readFileSync(path.resolve(args[2]));
 en = JSON.parse(en);
 
+// get the correct resource file type based on transifex project/repo and resource
+const getResourceType = (project, resource) => {
+    if (project === 'scratch-website') {
+        // all the resources are KEYVALUEJSON
+        return 'KEYVALUEJSON';
+    }
+    if (project === 'scratch-legacy') {
+        // all the resources are po files
+        return 'PO';
+    }
+    if (project === 'scratch-editor') {
+        if (resource === 'blocks') {
+            return 'KEYVALUEJSON';
+        }
+        // everything else is CHROME I18N JSON
+        return 'CHROME';
+    }
+    if (project === 'scratch-videos') {
+        // all the resources are srt files
+        return 'SRT';
+    }
+    if (project === 'scratch-android') {
+        // all the resources are android xml files
+        return 'ANDROID';
+    }
+    if (project === 'scratch-resources') {
+        // all the resources are Chrome format json files
+        return 'CHROME';
+    }
+    process.stdout.write(`Error - Unknown resource type for:\n`);
+    process.stdout.write(`  Project: ${project}, resource: ${resource}\n`);
+    process.exit(1);
```

Comment:
Note, I left this as `process.exit(1)` as this suggests an unknown 'project' or repository, and I really wanted it to fail right away.